File: app/src/main/java/com/example/orderfood/adapter/GroupMemberAdapter.java
package com.example.orderfood.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.example.orderfood.R;
import com.example.orderfood.model.User;

import java.util.List;

public class GroupMemberAdapter extends RecyclerView.Adapter<GroupMemberAdapter.ViewHolder> {

    private final List<User> memberList;
    private final int currentUserId;
    private final int creatorId;
    private final OnRemoveMemberClickListener listener;

    public GroupMemberAdapter(List<User> memberList, int currentUserId, int creatorId, OnRemoveMemberClickListener listener) {
        this.memberList = memberList;
        this.currentUserId = currentUserId;
        this.creatorId = creatorId;
        this.listener = listener;
    }

    public interface OnRemoveMemberClickListener {
        void onRemoveMember(User member);
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_group_member, parent, false);
        return new ViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        User member = memberList.get(position);
        holder.nickname.setText(member.getNickname());

        // Only the group creator can remove other members
        if (currentUserId == creatorId && member.getId() != currentUserId) {
            holder.removeButton.setVisibility(View.VISIBLE);
            holder.removeButton.setOnClickListener(v -> {
                listener.onRemoveMember(member);
            });
        } else {
            holder.removeButton.setVisibility(View.GONE);
        }
    }

    @Override
    public int getItemCount() {
        return memberList.size();
    }

    public static class ViewHolder extends RecyclerView.ViewHolder {
        public final TextView nickname;
        public final Button removeButton;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            nickname = itemView.findViewById(R.id.tv_member_nickname);
            removeButton = itemView.findViewById(R.id.btn_remove_member);
        }
    }
} 

File: app/src/main/java/com/example/orderfood/adapter/GroupMemberAdapter.kt
package com.example.orderfood.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.orderfood.R
import com.example.orderfood.model.User

class GroupMemberAdapter(
    private val memberList: List<User>,
    private val currentUserId: Int,
    private val creatorId: Int,
    private val listener: OnRemoveMemberClickListener
) : RecyclerView.Adapter<GroupMemberAdapter.ViewHolder>() {

    interface OnRemoveMemberClickListener {
        fun onRemoveMember(member: User)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_group_member, parent, false)
        return ViewHolder(view)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val member = memberList[position]
        holder.nickname.text = member.nickname

        // Only the group creator can remove other members
        if (currentUserId == creatorId && member.id != currentUserId) {
            holder.removeButton.visibility = View.VISIBLE
            holder.removeButton.setOnClickListener {
                listener.onRemoveMember(member)
            }
        } else {
            holder.removeButton.visibility = View.GONE
        }
    }

    override fun getItemCount(): Int = memberList.size

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val nickname: TextView = itemView.findViewById(R.id.tv_member_nickname)
        val removeButton: Button = itemView.findViewById(R.id.btn_remove_member)
    }
} 